### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by takuseno** [takuseno/d3rlpy#2]
- [x] survey scikit-learn API
- [x] design basic interfaces
- [x] setup CI
- [x] add unit tests
- [x] implement basic algorithms
  - [x] implement DQN
  - [x] implement DDPG
  - [x] implement SAC
  - [x] implement TD3
- [x] decorator to convert ndarray to torch.tensor
- [x] implement logging for training
  - [x] implement tensorboard support
  - [x] implement original logger
- [x] separate unit tests and performance tests
- [ ] implement evaluation metrics
  - [ ] OPC
    - [ ] modify MDPDataset to hold success flag
  - [x] advantage
  - [x] TD
  - [x] environment evaluation
  - [x] value scale
  - [x] compare policy
  - [x] action diff
- [x] check scikit-learn compatibility
- [x] add datasets functions
- [x] rename project
- [x] add preprocessing
- [x] load from params.json
- [ ] add action scaling
- [ ] add benchmark
  - [x] implement dataset object
  - [x] cartpole dataset for unit test
    - [ ] add success flags
  - [x] pendulum dataset for unit test
    - [ ] add success flags
  - [x] implement Atari example
    - https://github.com/google/dopamine/blob/e7d780d7c80954b7c396d984325002d60557f7d1/dopamine/replay_memory/circular_replay_buffer.py#L680
    - [x] implement d4rl_atari
    - [x] memory-efficient channel stacking
  - [ ] implement D4RL example
  - [x] implement pybullet example
    - [x] train SAC on Ant, Hopper, HalfCheetah
    - [x] implement d4rl_pybullet
- [ ] implement batch RL algorithms
  - [ ] implement REM
  - [x] implement BCQ
    - [x] continuous version
    - [x] discrete version
  - [x] implement BEAR
    - [x] continuous  version
    - [ ] discrete version
  - [x] implement CQL
    - [x] continuous version
    - [x] discrete version
  - [ ] implement ABM
    - [ ] implement continuous version
    - [ ] implement discrete version
  - [ ] implement MOPO
    - [x] implement interface to change dataset while training
    - [x] implement continuous version
    - [ ] implement discrete version
    - [ ] implement image observation version
  - [ ] implement AWR
    - [ ] implement lambda-return on Episode and MiniBatchTransition
  - [ ] implement AWAC
- [x] check GPU behavior
- [ ] implement environment termination option
- [ ] Dockerfile
  - [ ] make Dockerfile
  - [ ] DockerHub
- [ ] deployment
  - [ ] training server
  - [ ] API client
  - [ ] C++ client
  - [ ] add predict_value
  - [x] add uncertainty of predict_value
- [ ] implement techniques
  - [x] QR value function
  - [x] implicit quantile network Q function
  - [x] fully parametrized quantile function
  - [ ] policy adaptation during deployment.
  - [ ] distributional policy gradients
  - [ ] random network augmentation
  - [ ] Image Augmentation Is All You Need
  - [x] multiple GPU support in cross validation
  - [x] implement SUNRISE
- [ ] implement inference mode
- [ ] document
  - [x] add docstrings
  - [x] setup sphinx
  - [x] add distributional RL documents
  - [ ] add document test
  - [ ] host document
  - [ ] build CI pipeline for automatic update
- [ ] articles
